### ISSUE TITLE
[Question]: Why knowledgeGraphIndex return only english KG?

### ISSUE THREAD

**Issue by JinSeoung-Oh** [run-llama/llama_index#8971]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I'm building GraphRAG with nebular graph.
I made a module for extracting triplet from korean chunk data
And pass it KnowledgeGraphIndex, and it return english KG not korean KG.
I don't want translate my korean triplet. I just want to make korean KG.

I searched NebularGraph just only support English, but I cannot find out.
Basically, NebularGarph Query is English(at least, keyword)
So, maybe, because of this my korean triplet translate to english

But I can not find out where my korean triplt translate to english
So, please let me know this part and what LLM used for this.

Thanks!

**Comment by JinSeoung-Oh** [comment]
Thanks, it is very helpful.
But.. a few minutes ago, I re-run my code
and knowledgeGraphIndex return korean KG, suddenly

I was re-build and re-run my code for a week, but every try just failed.
So, I just re-run original code that has already failed dozens of times to return korean KG
And it return korean KG!

I don't know why..
I want stable result. 
In this case, there is no guarantee that I will get the results I want next time..

**Comment by JinSeoung-Oh** [comment]
@dosu-bot 
Thanks, it is very helpful.
But.. a few minutes ago, I re-run my code
and knowledgeGraphIndex return korean KG, suddenly

I was re-build and re-run my code for a week, but every try just failed.
So, I just re-run original code that has already failed dozens of times to return korean KG
And it return korean KG!

I don't know why..
I want stable result. 
In this case, there is no guarantee that I will get the results I want next time..

**Comment by JinSeoung-Oh** [comment]
@dosu-bot

Yeach.. maybe you are right.
I think KnowledgeGraphIndex has some bug when I pass non-english

For example, this is the part of my korean triplet

[["파이-1.5", "공개한", "마이크로소프트"], ["파이-1.5", "이다", "경량 멀티모달 언어모델"], ["파이-1.5", "이해하고 답변할 수 있는", "텍스트"], ["파이-1.5", "이해하고 답변할 수 있는", "이미지"], ["파이-1.5", "갖춘", "매개변수"], ["파이-1.5", "갖춘", "경량 모델"], ["파이-1.5", "결과를 얻을 수 있다고 주장", "대형언어모델"], ["파이-1.5", "훈련에 사용한", "교과서"], ["파이-1.5", "사용해", "훈련"], ["파이-1.5", "사용해", "A100 GPU"], ["파이-1.5", "비교되지 않지만 작업의 효율성과 에너지 효율성이 높음", "GPT-4"], ["파이-1.5", "보완적인 관계에 있음", "GPT-4"], ["마이크로소프트", "배제하고", "오픈AI"], ["마이크로소프트", "만들 수 있는 가능성을 보여주는", "자체 AI 제품"], ["인공지능", "철학적인 논쟁", "뇌와 마음의 관계"], ["인공지능", "인간 뉴런 작동 원리 모델 제시", "워런 매컬러와 월커 피츠"], ["인공지능", "인간의 일을 할 수 있는지 의문", "컴퓨터"], ["겨울", "연구 중단", "인공지능 연구"], ["인공지능", "도입된 방법론", "딥러닝"], ["인공지능", "성능 압도", "대회"], ["인공지능", "학습에 필요한 데이터의 중요성", "학습 데이터"]]

In this, GPT-4 just connect with "파이-1.5"
But when KnowledgeGraphIndex build KG, GPT-4 connected with "A100 GPU", "교과서","워런 매컬러와 월커 피츠" and so on..
It is wrong information from my triplet..

So, it just my thought, we have to make custom method for building non-english KG..

Anyway, Thanks!

**Comment by JinSeoung-Oh** [comment]
@dosu-bot
Thanks.
If I will have question when I investigate retrievers, I will ask it
Thanks!
